### PR TITLE
adapt to mojo 0.4

### PR DESCRIPTION
**Pull request by rd4com**
hope it helps,
i also added parallel to:
   #Apply the temperature to the logits

https://docs.modular.com/mojo/changelog.html#v0.4.0-2023-10-05

### CODE REVIEW

File: llama2.mojo [hunk 1]
```diff
@@ -977,8 +974,10 @@ fn main() raises:
                 next_token = argmax(state.logits)
             else:
                 # Apply the temperature to the logits
-                for q in range(config.vocab_size):
+                @parameter
+                fn temp(q:Int):
```
Comment (tairov) [inline llama2.mojo:978]: It make no sense to parallelize single operation (division "/")
much better to vectorize, would you mind implement it ? 
Also do some benchmarks. It doesn't make sense to implement optimization if there is no difference in execution time or "tok/s" performance

**Comment by rd4com** [comment]
I implemented the vectorize, as for the benchmarks i have no idea how to do it,
i have mojo 0.4.0 right now,
any idea ?

**Comment by mikowals** [comment]
@rd4com , there are examples of how to benchmark in the [matmul notebook](https://docs.modular.com/mojo/notebooks/Matmul.html) in the Mojo documentation.  For this change doing a small benchmark that only tests the changed loop code at vocab_size of 32000 iterations would be interesting.  Also just running `mojo llama2.mojo stories15M.bin" 5-10 times from this PR branch and again from master branch and reporting the results is useful information.

**Comment by tairov** [comment]
@rd4com would you mind to run some benchmarks ? Or better let's focus on Mojo 0.4.0 migration, for this let's not include `vectorize` change for now. Feel free to send another PR.

**Comment by rd4com** [comment]
Ok, i updated the repo, do you want me to close this pr and create another one ?
please verify the code is running fine before
